modules/github_integration/app/models/github_pull_request.rb

```diff
@@ -57,8 +56,15 @@ class GithubPullRequest < ApplicationRecord
                         unless: :partial?
   validate :validate_labels_schema
 
-  scope :complete, -> { where(state: ['open', 'closed']) }
   scope :without_work_package, -> { left_outer_joins(:work_packages).where(work_packages: { id: nil }) }
+  scope :partial, -> {
+    where(body: nil,
+          comments_count: nil,
+          review_comments_count: nil,
+          additions_count: nil,
+          deletions_count: nil,
+          changed_files_count: nil)
+  }
```

Comment:
I may be wrong, but is this scope used anywhere (except tests)? if not, we can probably remove it :)

Reply:
It is still used in the upsert service https://github.com/opf/openproject/blob/dev/modules/github_integration/lib/open_project/github_integration/services/upsert_pull_request.rb#L49. Admittedly, I am not sure whether that distinction is really necessary. It seems to me that both the html_url as well as the id are identifiers. So I have been thinking about writing a method that takes whatever of the two to find a `GithubPullRequest`. If both are present, the preference would be on the id.

What do you, @tessi, think about this?

Reply:
> Admittedly, I am not sure whether that distinction is really necessary. It seems to me that both the html_url as well as the id are identifiers. So I have been thinking about writing a method that takes whatever of the two to find a GithubPullRequest. If both are present, the preference would be on the id.

good idea! it makes less overall complexity since we can get rid of the partial concept.